processor/isolationforestprocessor/processor.go

```diff
@@ -103,6 +104,10 @@ func newIsolationForestProcessor(config *Config, logger *zap.Logger) (*isolation
 	}
 
 	processor.updateTicker = time.NewTicker(updateFreq)
+
+	// Properly track the background goroutine with WaitGroup
+	processor.shutdownWG.Add(1)
+
 	go processor.modelUpdateLoop()
```

Comment:
```suggestion
	// Properly track the background goroutine with WaitGroup
	processor.shutdownWG.Add(1)
	go func() {
		defer processor.shutdownWG.Done()
		processor.modelUpdateLoop()
	}()
```

Bit of a style preference, but I find that keeping the Add and Done calls close together makes it easier to reason about.